### PR TITLE
wamr: Add a few options for esp32s3

### PR DESCRIPTION
**Pull request by yamt**
## Summary
The corresponding WAMR change:
https://github.com/bytecodealliance/wasm-micro-runtime/pull/2348

### CODE REVIEW

File: interpreters/wamr/Kconfig [hunk 1]
```diff
@@ -47,6 +47,16 @@ config INTERPRETERS_WAMR_AOT_QUICK_ENTRY
 		signature.
 		Enable this option will increase the size of runtime, ~8KB.
 
+config INTERPRETERS_WAMR_AOT_WORD_ALIGN_READ
+	bool "Make WAMR AOT loader use word-aligned read"
+	default y
+	depends on INTERPRETERS_WAMR_AOT && ARCH_CHIP_ESP32S3
```
Comment (xiaoxiang781216) [inline interpreters/wamr/Kconfig:53]: why check ARCH_CHIP_ESP32S3 in common code?

Comment (yamt) [reply interpreters/wamr/Kconfig:53]: because these things in wamr are specific to esp32s3 at least for now.

Comment (xiaoxiang781216) [reply interpreters/wamr/Kconfig:53]: please remove esp32s3 specific code from wamr and remove ARCH_CHIP_ESP32S3 from here.

Comment (yamt) [reply interpreters/wamr/Kconfig:53]: i'm not happy with the current structure either.
but i want to make it work first.

Comment (xiaoxiang781216) [reply interpreters/wamr/Kconfig:53]: can we remove ARCH_CHIP_ESP32S3 from patch? espressif is working on to make the change in wamr more general for other arch/chip too.

Comment (yamt) [reply interpreters/wamr/Kconfig]: ok. done.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**